code/modules/clothing/suits/miscellaneous.dm

```diff
@@ -958,7 +958,9 @@
 	if(linked_staff.faith >= 100)	//if the linked staff is fully recharged, do nothing
 		return
 
-	if(!(linked_staff in range(3, get_turf(src))))		//staff won't charge at range (to prevent it from being handed off / stolen and used)
+	// Do not allow the staff to recharge if it's more than 3 tiles away from the robe. If get_dist returns 0, the robe and the staff in the same tile.
+	var/staff_dist = get_dist(H, linked_staff)
+	if(staff_dist > 3 || staff_dist < 0)
```

Comment:
Why not just do `if(staff_dist <= 3)`?

Reply:
So in the DM refs for `get_dist` it says it will return -1 for certain error conditions, but I'm not sure what those error conditions are exactly. So I only want to cover 0 to 3 and not the -1.

Reply:
`get_dist() will return -1 in error conditions, such as when Loc1 and Loc2 are the same object. `

I don't think this is something to worry about. I'd go with what I suggested.